src/entities/addLiquidity/index.ts

```diff
@@ -21,6 +23,13 @@ export class AddLiquidity implements AddLiquidityBase {
     ): Promise<AddLiquidityQueryOutput> {
         this.inputValidator.validateAddLiquidity(input, poolState);
         switch (poolState.vaultVersion) {
+            case 0: {
+                const addLiquidity = new AddLiquidityCowAmm();
```

Comment:
Seems likely there will be a future version of CowAMM pools. Should we already start checking for the new version and throw if not supported?

Reply:
This is actually checking for the `vaultVersion` and not the `poolVersion`. 
It's set as zero because that's how the API is setting it for v1 pools, meaning they have no vault.
Newer poolVersions should not affect how this is implemented as long as the contract interface for getting token balance and totalShares doesn't change.

Reply:
Sorry, my comment wasn't very clear. I actually meant we should maybe be checking for `poolVersion` too. My reasoning was that there will likely be a CowAMM V2 (which might have a different interface) or a completely different non-vault amm which I assume would be vaultVersion 0 too. Not a big deal but I think throwing an error when its an unexpected version would potentially avoid problems.

Reply:
As we talked, lets leave as is.